### PR TITLE
[Term Entry] C# Strings .ToCharArray()

### LINKED ISSUE #5572: [Term Entry] C# Strings .ToCharArray()
### Reason/inspiration (optional)

We would like a new entry on the `.ToCharArray()` term under the Strings concept in C#. The entry should go in a new file under `content/c-sharp/concepts/strings/terms/tochararray/tochararray.md`.

The entry should include:

- An introduction to the concept
- A `Syntax` section that provides the syntax for the concept
- An `Example` section that provides an example demonstrating the concept in use
- A `Codebyte Example` section that provides a codebyte example demonstrating the concept in use

Please refer to the [term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md) and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by PragatiVerma18**
### Description

Add a new entry on the `.ToCharArray()` term under the Strings concept in C#. 

### Issue Solved

Closes #5572

### Type of Change

- [x] Adding a new entry
- [ ] Editing an existing entry (fixing a typo, bug, issues, etc)
- [ ] Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/c-sharp/concepts/strings/terms/tochararray/tochararray.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+---
+Title: '.ToCharArray()'
+Description: 'Converts a string into a character array in C#. Each character in the string becomes an element in the array.'
```
Comment (mamtawardhani) [inline content/c-sharp/concepts/strings/terms/tochararray/tochararray.md:3]:
```suggestion
Description: 'Converts a string into an array of characters.'
```

File: content/c-sharp/concepts/strings/terms/tochararray/tochararray.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  - 'Code Foundations'
+Tags:
+  - 'Strings'
+  - 'Methods'
+CatalogContent:
+  - 'learn-c-sharp'
+  - 'paths/computer-science'
+---
+
+The **`.ToCharArray()`** method in C# is used to convert a string into an array of characters. Each character in the string becomes an element in the resulting array, allowing for manipulation or iteration of individual characters.
```
Comment (mamtawardhani) [inline content/c-sharp/concepts/strings/terms/tochararray/tochararray.md:15]:
```suggestion
The **`.ToCharArray()`** method in C# converts a string into an array of characters, where each character in the string becomes an element in the resulting array. This allows for the manipulation or iteration of individual characters.
```

File: content/c-sharp/concepts/strings/terms/tochararray/tochararray.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+```pseudo
+char[] arrayName = stringName.ToCharArray();
+```
+
+- `stringName`: The string to be converted into a character array.
+- `arrayName`: The resulting array containing individual characters from the string.
+
+### Optional Parameters
+
+The `.ToCharArray()` method also allows for optional parameters to specify a substring to convert:
```
Comment (mamtawardhani) [inline content/c-sharp/concepts/strings/terms/tochararray/tochararray.md:28]:
```suggestion
The `.ToCharArray()` method can also take optional parameters to convert a specific substring into a character array:
```

File: content/c-sharp/concepts/strings/terms/tochararray/tochararray.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+```pseudo
+char[] substringArray = stringName.ToCharArray(startIndex, length);
+```
+
+- `startIndex`: The zero-based index where the substring starts.
+- `length`: The number of characters to include in the resulting array.
```
Comment (mamtawardhani) [inline content/c-sharp/concepts/strings/terms/tochararray/tochararray.md:35]:
```suggestion
- `length`: The number of characters to include in the resulting array, starting from `startIndex`.
```

File: content/c-sharp/concepts/strings/terms/tochararray/tochararray.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+Here's an example of using `.ToCharArray()` to split a string into individual characters:
+
+```cs
+string greeting = "Hello, World!";
+char[] charArray = greeting.ToCharArray();
+
+foreach (char c in charArray)
+{
+    Console.WriteLine(c);
+}
```
Comment (mamtawardhani) [inline content/c-sharp/concepts/strings/terms/tochararray/tochararray.md:50]:
```suggestion
using System;

class Program
{
    static void Main()
    {
        string greeting = "Hello, World!";
        char[] charArray = greeting.ToCharArray();

        foreach (char c in charArray)
        {
            Console.WriteLine(c);
        }
    }
}
```

File: content/c-sharp/concepts/strings/terms/tochararray/tochararray.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+### Converting a Substring to a Character Array
+
+```cs
+string phrase = "Convert this string!";
+char[] partialArray = phrase.ToCharArray(8, 4);
+
+foreach (char c in partialArray)
+{
+    Console.WriteLine(c);
+}
```
Comment (mamtawardhani) [inline content/c-sharp/concepts/strings/terms/tochararray/tochararray.md:80]:
```suggestion
using System;

class Program
{
    static void Main()
    {
        string phrase = "Convert this string!";
        char[] partialArray = phrase.ToCharArray(8, 4);

        foreach (char c in partialArray)
        {
            Console.WriteLine(c);
        }
    }
}
```

File: content/c-sharp/concepts/strings/terms/tochararray/tochararray.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+```shell
+t
+h
+i
+s
+```
+
+## Codebyte Example
+
+Try this example to see how `.ToCharArray()` works in C#:
```
Comment (mamtawardhani) [inline content/c-sharp/concepts/strings/terms/tochararray/tochararray.md:94]:
```suggestion
Run the following codebyte example to see how `.ToCharArray()` works in C#:
```

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @PragatiVerma18 , thank you for contributing to Codecademy Docs, the entry is nicely written! 😄 

I've suggested a few changes, could you please review and modify those at your earliest convenience? Thank you! 😃

**Comment by PragatiVerma18** [comment]
Hey @mamtawardhani, I have pushed the changes. Thanks for the review and feedback. 🚀

**Review by mamtawardhani — APPROVED** [review]
Thank you for contributing to Codecademy Docs @PragatiVerma18  😄 

The entry looks good to be merged! 🚀

**Pull request merged**